src/careamics/dataset_ng/factory.py

```diff
@@ -112,8 +110,32 @@ def init_patch_extractor(
     return patch_extractor(image_stacks)
 
 
-def select_patch_extractor_type(in_memory: bool) -> type[PatchExtractor]:
-    if in_memory:
+def select_patch_extractor_type(
+    data_type: SupportedData,
+    in_memory: bool,
+) -> type[PatchExtractor]:
+    """Select the appropriate PatchExtractor type based on data type and memory mode.
+
+    If `in_memory` is True, or `data_type` is ZARR or CZI, the standard
+    `PatchExtractor` is selected, otherwise the `LimitFilesPatchExtractor` will be used.
+
+    Parameters
+    ----------
+    data_type : SupportedData
+        The type of data being handled.
+    in_memory : bool
+        Indicates whether data is to be loaded into memory.
+
+    Returns
+    -------
+    type[PatchExtractor]
+        The selected PatchExtractor type.
+    """
+    if in_memory or data_type in (
+        SupportedData.ZARR,
+        SupportedData.CZI,
+        SupportedData.ARRAY,
+    ):
         return PatchExtractor
     else:
         return LimitFilesPatchExtractor
```

Comment:
mm, sorry if this is annoying but maybe it's better to invert the condition, to me it's clearer to see the cases this way around:
```suggestion
    if not in_memory and data_type in (SupportedData.TIFF, SupportedData.CUSTOM):
        return LimitFilesPatchExtractor
    else:
        return PatchExtractor
```

Reply:
No it makes sense, thanks!